core/server/services/auth/passwordreset.js

```diff
@@ -0,0 +1,38 @@
+const _ = require('lodash');
+const security = require('../../lib/security');
+const constants = require('../../lib/constants');
+const common = require('../../lib/common');
+const models = require('../../models');
+
+function generateToken(email, settingsAPI) {
+    const options = {context: {internal: true}};
+    let dbHash, token;
+
+    return settingsAPI.read(_.merge({key: 'db_hash'}, options))
+        .then((response) => {
+            dbHash = response.settings[0].value;
+
+            return models.User.getByEmail(email, options);
+        })
+        .then((user) => {
+            if (!user) {
+                throw new common.errors.NotFoundError({message: common.i18n.t('errors.api.users.userNotFound')});
+            }
+
+            token = security.tokens.resetToken.generateHash({
```

Comment:
Okay I realise this isn't part of your PR but the details for creating this hash seem a bit mental? Why do we need a uuid in there & and the hashed password!?

Reply:
From the perspective of outside use of this token, we only need to have `email` and `expires` bits. The rest of the fields doesn't seem to help in any way by being there :man_shrugging: Think it's worth having a deeper investigation into it and strip needed data to a bare minimum. Might be a thing to check when doing:
> @TODO: use brute force middleware (see https://github.com/TryGhost/Ghost/pull/7579)